kolibri/auth/test/test_permissions.py

```diff
@@ -1,454 +1,568 @@
+        self.assertFalse(unsaved_deviceowner.user_can_create(self.member))
+        self.assertFalse(unsaved_deviceowner.user_can_create(self.data["unattached_users"][0]))
+
+    def test_non_device_owners_cannot_read_device_owner(self):
+        """ Users who are not DeviceOwners cannot read a DeviceOwner """
+        self.assertFalse(self.device_owner.user_can_read(self.data["facilityadmin"]))
+        self.assertFalse(self.device_owner.user_can_read(self.data["facilitycoach"]))
+        self.assertFalse(self.device_owner.user_can_read(self.own_classroom_admin))
+        self.assertFalse(self.device_owner.user_can_read(self.own_classroom_coach))
+        self.assertFalse(self.device_owner.user_can_read(self.member))
+        self.assertFalse(self.device_owner.user_can_read(self.data["unattached_users"][0]))
+
+    def test_non_device_owners_cannot_update_device_owner(self):
+        """ Users who are not DeviceOwners cannot update a DeviceOwner """
+        self.assertFalse(self.device_owner.user_can_update(self.data["facilityadmin"]))
+        self.assertFalse(self.device_owner.user_can_update(self.data["facilitycoach"]))
+        self.assertFalse(self.device_owner.user_can_update(self.own_classroom_admin))
+        self.assertFalse(self.device_owner.user_can_update(self.own_classroom_coach))
+        self.assertFalse(self.device_owner.user_can_update(self.member))
+        self.assertFalse(self.device_owner.user_can_update(self.data["unattached_users"][0]))
+
+    def test_non_device_owners_cannot_delete_device_owner(self):
+        """ Users who are not DeviceOwners cannot delete a DeviceOwner """
+        self.assertFalse(self.device_owner.user_can_delete(self.data["facilityadmin"]))
+        self.assertFalse(self.device_owner.user_can_delete(self.data["facilitycoach"]))
+        self.assertFalse(self.device_owner.user_can_delete(self.own_classroom_admin))
+        self.assertFalse(self.device_owner.user_can_delete(self.own_classroom_coach))
+        self.assertFalse(self.device_owner.user_can_delete(self.member))
+        self.assertFalse(self.device_owner.user_can_delete(self.data["unattached_users"][0]))
+
+    def test_device_owner_can_do_anything_to_a_device_owner(self):
+        """ DeviceOwner can do anything to a DeviceOwner, except delete itself """
+        self.assertTrue(self.device_owner2.user_can_create(self.device_owner))
+        self.assertTrue(self.device_owner2.user_can_read(self.device_owner))
+        self.assertTrue(self.device_owner2.user_can_update(self.device_owner))
+        self.assertTrue(self.device_owner2.user_can_delete(self.device_owner))
+        self.assertTrue(self.device_owner.user_can_create(self.device_owner))
+        self.assertTrue(self.device_owner.user_can_read(self.device_owner))
+        self.assertTrue(self.device_owner.user_can_update(self.device_owner))
+        self.assertFalse(self.device_owner.user_can_delete(self.device_owner))
```

Comment:
>can_do_anything

>self.assertFalse

These are inconsistent. Someone reading the test without more context might imagine that this is an error. Do you find that problematic? If so, how could we address it?